win32ss/user/user32/misc/dllmain.c

```diff
@@ -465,6 +466,19 @@ Cleanup(VOID)
     DeleteFrameBrushes();
 }
 
+static HRESULT
+GetImmFileName(_Out_ LPWSTR lpBuffer,
+               _In_ size_t cchBuffer)
+{
+    UINT length = GetSystemDirectoryW(lpBuffer, cchBuffer);
+    if (length && length < cchBuffer)
+    {
+        StringCchCatW(lpBuffer, cchBuffer, L"\\");
+        return StringCchCatW(lpBuffer, cchBuffer, L"imm32.dll");
+    }
+    return StringCchCopyW(lpBuffer, cchBuffer, L"imm32.dll");
+}
```

Comment:
Can you remove that function, and make the other one (that is already in `misc/imm32.c`) non-static and available in a header? (much like the other function `InitializeImmEntryTable` is?)

Reply:
Why?
If this function is only used in once place, why move it into a header?

Reply:
It's used in two places: here (`misc/dllmain.c`) **and** in `misc/imm32.c`, but also defined twice in these same places.

Reply:
Fixed.